rust/src/mime/mime.rs

```diff
@@ -0,0 +1,605 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use crate::common::nom7::take_until_and_consume;
+use nom7::branch::alt;
+use nom7::bytes::complete::{tag, take, take_till, take_until, take_while};
+use nom7::character::complete::char;
+use nom7::combinator::{complete, opt, rest, value};
+use nom7::error::{make_error, ErrorKind};
+use nom7::{Err, IResult};
+use std;
+use std::collections::HashMap;
+
+#[derive(Clone)]
+pub struct HeaderTokens<'a> {
+    pub tokens: HashMap<&'a [u8], &'a [u8]>,
+}
+
+fn mime_parse_value_delimited(input: &[u8]) -> IResult<&[u8], &[u8]> {
+    let (input, _) = char('"')(input)?;
+    let mut escaping = false;
+    for i in 0..input.len() {
+        if input[i] == b'\\' {
+            escaping = true;
+        } else {
+            if input[i] == b'"' && !escaping {
+                return Ok((&input[i + 1..], &input[..i]));
```

Comment:
Could unicode mess with this.  I recently ran into this: https://github.com/jasonish/rust-suricatax-rule-parser/commit/d77c4e3ed20f0dce054bb9911b42c4efe13f80ff

Reply:
This is a buffer `&[u8]` not a string, so there is no need to worry about unicode, right ?